### DISCUSSION TITLE
Route Model Binding In Data Objects When Using For Request

### DISCUSSION THREAD

**Discussion by gurpreetkaits** [spatie/laravel-data#805] [Ideas]
Hi everyone,

For many days I've been working with Laravel-Data, I had a question when we map route properties to DTO properties.

`#[FromRouteParameter('product')]`
` public Product $product`
 Something as mentioned above, how can we take advantage of Route model binding for Product?

I tried but it wasn't working, it says string passed, required type ***/model

**Comment by toniette** [comment]
I'm facing the same issue here. I think having models inside your DTO isn't a good practice, because doing it causes a higher coupling level with other app layers, but even that, It would be nice to have this feature. Maybe creating a custom Attribute and applying it solves our issue. I'm trying to figure out that right now.
